### PR TITLE
Extrapolate look ahead points to prevent rapid turning when approaching final target way point

### PR DESCRIPTION
**Pull request by PhilipAxelrod**
Rather than restricting the look ahead point to be on the defined path, we instead extrapolate the line segments, and use the look ahead point on the extrapolated line. This fixes the bug where the robot turns 90 degrees upon reaching the target turns 90 degrees, because instead of turning toward the last point in the path, the robot turns toward an imaginary point ahead if itself.

### CODE REVIEW

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala [hunk 1]
```diff
@@ -80,99 +81,98 @@ trait PurePursuitControllers {
   /**
     * returns an angle limited from -180 to 180 equivalent to the input
     * @param degrees
-    * @return angle betwee -270 and 90
+    * @return angle betwee -90 and 90
```
Comment (shadaj) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala:84]: This should be -180 to 180, right?

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala [hunk 2]
```diff
@@ -42,7 +42,8 @@ trait PurePursuitControllers {
     * @param lookAheadDistance
     * @return the look ahead point for the given path
     */
-  def getLookAheadPoint(biSegmentPath: (Segment, Option[Segment]),
+  @Deprecated
```
Comment (shadaj) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala]: Use the lowercase `@deprecated`

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleCoreControllers.scala [hunk 3]
```diff
@@ -122,4 +122,32 @@ trait UnicycleCoreControllers {
...
+    val control = PIDF.pid(
+      hardware.turnPosition.toPeriodic,
+      targetAbsolute.toPeriodic,
+      props.map(_.turnPositionControlGains)
+    ).map(s => UnicycleSignal(Percent(0), s))
+
+    (control, error)
+  }
+
+  def turnPositionControl(targetAbsolute: PeriodicSignal[Angle])
```
Comment (shadaj) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleCoreControllers.scala]: Why is this method duplicated 3 times?

**Review by shadaj — CHANGES_REQUESTED** [review]
Couple of changes and we should be able to merge!

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleCoreControllers.scala (hunk 3 above)
Comment (PhilipAxelrod) [reply commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleCoreControllers.scala]: Yeah, I'll remove these

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala (hunk 1 above)
Comment (PhilipAxelrod) [reply commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala:84]: I deal with the case that the target angle is 180 or similar by calling this method. So if the input angle is 180, it returns 0. You are right that it is not returning an equivalent angle, but that is compensated by the fact that we drive backwards

**Review by shadaj — APPROVED** [review]
LGTM

**Pull request merged**